bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Executor.scala

```diff
@@ -0,0 +1,42 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.spModel.config2.{Config, ConfigSequence, ItemKey}
+import edu.gemini.spModel.obscomp.InstConstants.INSTRUMENT_NAME_PROP
+import edu.gemini.spModel.seqcomp.SeqConfigNames.INSTRUMENT_KEY
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+ * Created by jluhrs on 4/28/15.
+ */
+final class Executor(sequenceConfig: ConfigSequence) {
+
+  def run: Future[Int] = runSequence(sequenceConfig.getAllSteps.toList)
+
+  private def runSequence(config: List[Config]): Future[Int] = {
+    config match {
+      case Nil => Future.failed(new Exception("Empty sequence configuration"))
```

Comment:
I agree with @swalker2m.  I think one of the following is true:
- There is no logical problem with executing an empty sequence; it is simply a no-op, or
- An empty sequence makes no sense, in which case `type Sequence = NonEmptyList[Config]` and your program will not compile unless you can prove your sequence is non-empty.